src/plone/api/content.py

```diff
@@ -67,7 +70,18 @@ def create(
     :Example: :ref:`content-create-example`
     """
     # Create a temporary id if the id is not given
-    content_id = not safe_id and id or str(random.randint(0, 99999999))
+    if not safe_id and id:
+        content_id = id
+    else:
+        # Try to generate a unique random ID
+        attempts = 0
+        while attempts < MAX_UNIQUE_ID_ATTEMPTS:
+            content_id = str(random.randint(0, 99999999))
+            if content_id not in container:
+                break
+            attempts += 1
+        # If we couldn't find a unique ID after max attempts, use the last one anyway
+        # This maintains the original behavior if all attempts fail
```

Comment:
Here we could even decide to raise an exception, I will leave this decision to other reviewers

Reply:
I would indeed say it is better to raise an explicit error here, instead of encountering a less understandable error later on.  According to issue #445 you would get a BadRequest error.  But there is nothing wrong with the request, so we should use something else.  Maybe:

```
raise ValueError("Could not find unique id while creating content."
```

Reply:
Makes sense!